plugins/readtimer.koplugin/main.lua

```diff
@@ -50,6 +86,89 @@ function ReadTimer:unschedule()
     end
 end
 
+function ReadTimer:eyeSaverScheduleTimeCompute()
+    self.eyesaver.display_timestamp = os.time() + self.eyesaver.display_interval
+    self.eyesaver.display_time_clock = os.date(_("%H:%M"), self.eyesaver.display_timestamp)
+    self:eyeSaverSaveSettings()
+end
+
+function ReadTimer:eyeSaverSchedule()
+    if not self.eyesaver.enabled then
+        return
+    end
+    if not self:eyeSaverMessageScheduled() then
+        self:eyeSaverScheduleTimeCompute()
+    end
+    UIManager:scheduleIn(self.eyesaver.display_interval, self.eyesaver.timed_display_callback)
+end
+
+-- when time for displaying the message was in a period in which the ereader "slept", adapt the new display time to be 20 minutes in the future again:
+function ReadTimer:eyeSaverScheduleAdapt()
+    if not self.eyesaver.enabled then
+        return
+    end
+    if self.eyesaver.display_timestamp and self:eyeSaverMessageScheduled() and self.eyesaver.display_timestamp < os.time() then
+        self:eyeSaverUnschedule();
+        self:eyeSaverScheduleTimeCompute()
+        self:eyeSaverSchedule();
+    end
+end
+
+function ReadTimer:eyeSaverUnschedule()
+    if self:eyeSaverMessageScheduled() then
+        UIManager:unschedule(self.eyesaver.timed_display_callback)
+    end
+    self:eyeSaverResetTimer()
+end
+
+function ReadTimer:eyeSaverMessageScheduled()
+    return self.eyesaver.display_timestamp > 0
+end
+
+function ReadTimer:eyeSaverResetTimer()
+    self.eyesaver.display_timestamp = 0
+    self.eyesaver.display_time_clock = ''
+    self:eyeSaverSaveSettings()
+end
+
+function ReadTimer:eyeSaverSaveSettings()
+    self.eyesaver.settings:saveSetting("eyesaver_timestamp", self.eyesaver.display_timestamp)
```

Comment:
I guess you only need to save it as a setting because the next timer would be lost when you switch books or between reader and filemanager? As when you restart the reader, one should be fine to restart the timer to have the message in 20mn since the restart ?
In which case, you could just save it into another long-living table, like PluginShare (which is already used to keep some state across plugin reload).